### ISSUE TITLE
Wrapping cursor causes crash

### ISSUE THREAD

**Issue by st3iny** [alacritty/alacritty#3960]
1. Generate some output (e.g. run `ls -hal`)
2. Input spaces until the cursor wraps to the next line
3. Increase the window width until the cursor no longer wraps to the next line
4. Reduce window width until cursor wraps again
5. Alacritty crashes

This could be related to #3910.

### System

OS: Linux
Version: alacritty 0.5.0-dev (18cf806a)
Linux: X11, dwm (no compositor)

### Logs

`alacritty -vv` yields:
```
Created log file at "/tmp/Alacritty-103543.log"
[2020-07-13 11:42:21.389807902] [INFO] Welcome to Alacritty
[2020-07-13 11:42:21.389857596] [INFO] Configuration loaded from "/home/richard/.config/alacritty.yml"
[2020-07-13 11:42:21.397684584] [DEBUG] Estimated DPR: 1.25
[2020-07-13 11:42:21.397697348] [DEBUG] Estimated Cell Size: 11 x 22
[2020-07-13 11:42:21.397701456] [DEBUG] Estimated Dimensions: None
[2020-07-13 11:42:21.440272952] [INFO] Device pixel ratio: 1.25
[2020-07-13 11:42:21.441475658] [INFO] Initializing glyph cache...
[2020-07-13 11:42:21.450842083] [INFO] ... finished initializing glyph cache in 0.009355525s
[2020-07-13 11:42:21.450853775] [INFO] Cell Size: 11 x 22
[2020-07-13 11:42:21.450857993] [INFO] Padding: 9 x 14
[2020-07-13 11:42:21.450864475] [INFO] Width: 800, Height: 600
[2020-07-13 11:42:21.453132429] [INFO] PTY dimensions: Line(26) x Column(71)
[2020-07-13 11:42:21.456199581] [INFO] Initialisation complete
[2020-07-13 11:42:21.456490957] [DEBUG] New num_cols is 90 and num_lines is 62
[2020-07-13 11:42:21.464921938] [INFO] Width: 1012, Height: 1393
[2020-07-13 11:42:27.131617410] [DEBUG] New num_cols is 106 and num_lines is 62
[2020-07-13 11:42:27.131749648] [INFO] Width: 1184, Height: 1393
[2020-07-13 11:42:27.587766803] [DEBUG] New num_cols is 90 and num_lines is 62
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 18446744073709551615', /home/richard/.cache/yay/alacritty-git/src/alacritty/alacritty_terminal/src/grid/resize.rs:276:20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'PTY reader' panicked at 'index out of bounds: the len is 0 but the index is 4', /home/richard/.cache/yay/alacritty-git/src/alacritty/alacritty_terminal/src/term/mod.rs:1902:34
```

**Comment by chrisduerr** [comment]
> Input spaces until the cursor wraps to the next line

Note that this doesn't seem to be necessary. Just inputting a bunch of spaces without moving it to the next line and then shrinking the window size seems sufficient.

**Comment by chrisduerr** [comment]
This should be fixed in #3966. If you're interested in testing it, please let me know if it works.

**Comment by st3iny** [comment]
Works fine now. Thanks for the quick fix.
